roles/cluster_deploy_catalog_source/tasks/main.yml

```diff
@@ -0,0 +1,56 @@
+---
+- name: Fail if catalog_image is undefined
+  fail:
+    msg: Must provide a catalog_image
+  when: catalog_image is undefined
+
+- name: Fail if catalog_registry is undefined
+  fail:
+    msg: Must provide a catalog_registry
+  when: catalog_registry is undefined
+
+- name: Set default catalog_version (N-1) if undefined
+  set_fact:
+    catalog_version: "v{{ openshift_release_major }}.{{ openshift_release_minor|int - 1 }}"
+  when: catalog_version is undefined
+
+- name: Set catalog_name
+  set_fact:
+    catalog_name: "{{ catalog_image }}-cs"
+
+
+- name: 'Deploy {{ catalog_image }}:{{ catalog_version }} CatalogSource'
+  block:
+  - name: Catalog source version
+    debug:
+      msg: "Current OCP version: {{ openshift_release }}, Setting catalog tag {{ catalog_version }}"
+  
+  - name: Template CatalogSource for previous OCP release
+    template:
+      src: "{{ operator_catalog_previous }}"
+      dest: "{{ artifact_extra_logs_dir }}/{{ catalog_image }}-catalog-previous.yml"
+      mode: 0400
+  
+  - name: Apply CatalogSource for previous OCP release
+    command: oc apply -f "{{ artifact_extra_logs_dir }}/{{ catalog_image }}-catalog-previous.yml"
+    changed_when: true
```

Comment:
the word "previous" is not really relevant anymore, since we can deploy from any image, but that's just nit picking

Reply:
Yep - this is true, and something I missed. Thanks!

Reply:
is the `changed_when: true` also coming from the linter?
as this repo isn't using Ansible the conventional way, it doesn't make sense to track `changed` actions I would say

Reply:
Yes and no. Usually it wants me to add `changed` on commands and shells, I guess it depends. Here I added it out of habit. removed.